### PR TITLE
[BugFix] Fix ConsistencyChecker error dropping tablet if creating table takes too long

### PR DESCRIPTION
**Pull request by gengjun-git**
## Why I'm doing:
When creating a table, the tabletMeta will be put into TabletInvertedIndex first, and when all tablets are created on backend, the table/partition is put into local metastore. If the creation of tablet on backend costs too long, the tablet will be deleted by 
ConsistencyChecker.

## What I'm doing:
Record the id of the table being created into the ConsistencyChecker and ignore the check of the tablet of the table being created.

Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by HangyuanLiu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/consistency/ConsistencyChecker.java [hunk 1]
```diff
@@ -459,4 +462,18 @@ public void addTabletsToCheck(List<Long> tabletIds) {
...
         }
     }
+
+    public void addCreatingTableId(long tableId) {
+        creatingTableIds.compute(tableId, (k, v) -> (v == null) ? 1 : v + 1);
+    }
+
+    public void deleteCreatingTableId(long tabletId) {
+        creatingTableIds.compute(tabletId, (k, v) -> {
+            if (v == null || v <= 1) {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/consistency/ConsistencyChecker.java:472]: I don't think it's safe to protect the logic here with only concurrent hashmap, better using synchronized(creatingTableIds) {...}?

File: fe/fe-core/src/main/java/com/starrocks/consistency/ConsistencyChecker.java [hunk 2]
```diff
@@ -95,6 +96,8 @@ public class ConsistencyChecker extends FrontendDaemon {
     private int endTime;
     private long lastTabletMetaCheckTime = 0;
 
+    private final Map<Long, Integer> creatingTableIds = new ConcurrentHashMap<>();
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/consistency/ConsistencyChecker.java:99]:
```suggestion
    private final Map<Long, Integer> creatingTableCounters = new ConcurrentHashMap<>();
```
also you need to explain this structure, it's kind of hard to understand.

File: fe/fe-core/src/main/java/com/starrocks/consistency/ConsistencyChecker.java (hunk 1 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/consistency/ConsistencyChecker.java:472]: image
It is atomically, and I have tried.

File: fe/fe-core/src/main/java/com/starrocks/consistency/ConsistencyChecker.java (hunk 2 above)
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/consistency/ConsistencyChecker.java]: done

**Review by HangyuanLiu — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Pull request merged**